provisioner/src/lib.rs

```diff
@@ -522,3 +860,153 @@ fn engine_to_port(engine: aws_rds::Engine) -> String {
         aws_rds::Engine::Mysql(_) => "3306".to_string(),
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::time::Duration;
+
+    use aws_sdk_dynamodb::types::{
+        AttributeDefinition, KeySchemaElement, KeyType, ProvisionedThroughput, ScalarAttributeType,
+    };
+    use tokio::time::sleep;
+
+    use crate::MyProvisioner;
+
+    use shuttle_common::delete_dynamodb_tables_by_prefix;
+
+    async fn make_test_provisioner() -> MyProvisioner {
+        let pg_uri = "postgres://postgres:password@localhost:5432".to_string();
+        let mongo_uri = "mongodb://mongodb:password@localhost:8080".to_string();
+
+        MyProvisioner::new(
+            &pg_uri,
+            &mongo_uri,
+            "fqdn".to_string(),
+            "pg".to_string(),
+            "mongodb".to_string(),
+        )
+        .await
+        .unwrap()
+    }
+
+    async fn create_dynamodb_table(dynamodb_client: &aws_sdk_dynamodb::Client, table_name: &str) {
+        let attribute_definition = AttributeDefinition::builder()
+            .attribute_name("test")
+            .attribute_type(ScalarAttributeType::S)
+            .build();
+
+        let key_schema = KeySchemaElement::builder()
+            .attribute_name("test")
+            .key_type(KeyType::Hash)
+            .build();
+
+        let provisioned_throughput = ProvisionedThroughput::builder()
+            .read_capacity_units(10)
+            .write_capacity_units(5)
+            .build();
+
+        dynamodb_client
+            .create_table()
+            .table_name(table_name)
+            .key_schema(key_schema.clone())
+            .attribute_definitions(attribute_definition.clone())
+            .provisioned_throughput(provisioned_throughput.clone())
+            .send()
+            .await
+            .unwrap();
+    }
+
+    #[tokio::test]
+    async fn test_create_and_delete_dynamodb_policy() {
+        let provisioner = make_test_provisioner().await;
+
+        let prefix = provisioner.get_prefix("my_cool_project").await;
+
+        provisioner.create_dynamodb_policy(&prefix).await.unwrap();
+
+        provisioner.delete_dynamodb_policy(&prefix).await.unwrap();
+    }
```

Comment:
Nice!!! Will they also run successfully when a machine / CI does not have AWS credentials set? In which case the `ignore` attribute just has to be added.

Reply:
Yep, this test and the following ones don't work without AWS credentials. I'll `ignore` them.

Reply:
Done